frontend/ui/uimanager.lua

```diff
@@ -369,6 +369,8 @@ function UIManager:close(widget, refreshtype, refreshregion, refreshdither)
         for i = 1, #self._window_stack do
             self:setDirty(self._window_stack[i].widget)
         end
+        -- Inform everyone who's the boss now.
+        self:broadcastEvent(Event:new("TopWidget", ((self._window_stack[#self._window_stack] or {}).widget or {}).name))
```

Comment:
@poire-z @NiLuJe Feels a bit dirty but it seems to work. Any suggestions on a better way? Instead of unscheduling the plugin on an event there could also be a method that's checked before turning the page or something.

Reply:
How much do we end up firing that `TopWidget` event, ultimately?

Reply:
> How much do we end up firing that TopWidget event

Each time we UIManager:show() a widget, which is not that much, but it goes thru all the widgets to broadcast it. Which might be negligeable, I don't know, but still :)

> Instead of unscheduling the plugin on an event there could also be a method that's checked before turning the page or something.

Like I suggested above?:
> Or just a simple method UIManager:getTopWidgetId() => "ReaderUI" or "#1234ab32" that the widget could just call before deciding to send the onGotoViewRel.

Reply:
(Why did you remove the one on :show() and kept only the one on :close() ? )

Reply:
> How much do we end up firing that TopWidget event, ultimately?

Not much at all. So it's distributing the event to all widgets every once in a while vs always checking the top widget every x seconds. But of course this event implementation would affect everything and the plugin just users of the plugin, so out the event will go. ;-)

> Why did you remove the one on :show() and kept only the one on :close() ? )

It wasn't working right, ran into some kind of loop.

Reply:
(Well, the event could have some other usages, like stopping statistics, to get a more accurate time to read a page.
Unless all the little times we spend dict or wikipedia lookup'ing need to be accounted in this, assuming it's part of the reading time :)

Reply:
I don't know, I think the statistics are pretty cool but I'm not very concerned about their accuracy personally.

Reply:
@poire-z Btw, that issue I noticed with it continuously popping up end of document dialogs was magically resolved by your suggestion, because now it automatically pauses when that dialog is up. :-)